### PR TITLE
PDI-15038 Support MongoDB AuthenticationDatabase Parameter in MongoDB…

### PR DESCRIPTION
**Pull request by flatrixnet**
… Steps and Support the SCRAM-SHA-1 security

### CODE REVIEW

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.4 -Djunit.jvmargs=-XX:PermSize=256m -f build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  build.properties
  mongo-security-ext.MF
  out/production/pentaho-mongo-utils/org/pentaho/mongo/messages/messages.properties
  out/production/pentaho-mongo-utils/org/pentaho/mongo/wrapper/messages/messages.properties
  src/org/pentaho/mongo/MongoProp.java
  src/org/pentaho/mongo/wrapper/UsernamePasswordMongoClientWrapper.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File src/org/pentaho/mongo/wrapper/UsernamePasswordMongoClientWrapper.java
1. Message: '(' is not followed by whitespace. at Line: 67 Column: 8
2. Message: ')' is not preceded with whitespace. at Line: 67 Column: 39
3. Message: ')' is not preceded with whitespace. at Line: 67 Column: 64
4. Message: 'if' is not followed by whitespace. at Line: 67 Column: 7
5. Message: '{' at column 5 should be on the previous line. at Line: 68 Column: 5
6. Message: ')' is not preceded with whitespace. at Line: 71 Column: 37
7. Message: '}' at column 5 should be on the same line as the next part of a multi-block statement. at Line: 73 Column: 5
8. Message: ')' is not preceded with whitespace. at Line: 77 Column: 37



Newly Broken Tests:
-----------------
org.pentaho.mongo.functional.ClientWrapperIT.message: 
```

```
org.pentaho.mongo.functional.ClientWrapperIT.type: 
```

```
org.pentaho.mongo.functional.ClientWrapperIT.severity: 
```

```



Still Broken Tests:
-----------------
org.pentaho.mongo.functional.ClientWrapperIT.testCursor[1]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain password from user

	at com.sun.security.auth.module.Krb5LoginModule.promptForPass(Krb5LoginModule.java:897)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:760)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[1]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain password from user

	at com.sun.security.auth.module.Krb5LoginModule.promptForPass(Krb5LoginModule.java:897)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:760)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCreateDropCollection[7]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCreateDropCollection[3]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCreateDropCollection[2]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[0]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain Principal Name for authentication 
	at com.sun.security.auth.module.Krb5LoginModule.promptForName(Krb5LoginModule.java:841)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:704)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[5]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCursor[7]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCreateDropCollection[1]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain password from user

	at com.sun.security.auth.module.Krb5LoginModule.promptForPass(Krb5LoginModule.java:897)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:760)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCreateDropCollection[0]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain Principal Name for authentication 
	at com.sun.security.auth.module.Krb5LoginModule.promptForName(Krb5LoginModule.java:841)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:704)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[7]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[6]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[3]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testReadPref[4]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCursor[2]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCursor[3]: java.lang.NullPointerException
```
java.lang.NullPointerException
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.getCredentialList(UsernamePasswordMongoClientWrapper.java:67)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.getClient(NoAuthMongoClientWrapper.java:145)
	at org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper.<init>(NoAuthMongoClientWrapper.java:77)
	at org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper.<init>(UsernamePasswordMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:48)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)

```
org.pentaho.mongo.functional.ClientWrapperIT.testCursor[0]: org.pentaho.mongo.MongoDbException
```
org.pentaho.mongo.MongoDbException: Unable to authenticate as 'buildguy@PENTAHO.QA'
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:69)
	at org.pentaho.mongo.wrapper.KerberosMongoClientWrapper.<init>(KerberosMongoClientWrapper.java:45)
	at org.pentaho.mongo.wrapper.MongoClientWrapperFactory.createMongoClientWrapper(MongoClientWrapperFactory.java:46)
	at org.pentaho.mongo.functional.ClientWrapperIT.getWrapper(ClientWrapperIT.java:292)
	at org.pentaho.mongo.functional.ClientWrapperIT.runAfter(ClientWrapperIT.java:71)
Caused by: javax.security.auth.login.LoginException: Unable to obtain Principal Name for authentication 
	at com.sun.security.auth.module.Krb5LoginModule.promptForName(Krb5LoginModule.java:841)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:704)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.pentaho.mongo.KerberosUtil.loginAs(KerberosUtil.java:216)
	at org.pentaho.mongo.KerberosHelper.login(KerberosHelper.java:67)

```

Unit test coverage change:
============
org.pentaho.mongo.wrapper.MongoClientWrapperFactory:
 * instructionPercentCoverageChange :**-32.08%**
 * branchPercentCoverageChange :**-33.33%**
 * linePercentCoverageChange :**-25.00%**
 * complexityPercentCoverageChange :**-16.67%**
    
org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper:
 * instructionPercentCoverageChange :**-77.31%**
 * linePercentCoverageChange :**-71.80%**
 * complexityPercentCoverageChange :**-55.00%**
 * methodPercentCoverageChange :**-50.00%**
    
org.pentaho.mongo.MongoProp:
 * instructionPercentCoverageChange : +0.18%
    
Integration test coverage change:
============
org.pentaho.mongo.wrapper.MongoClientFactory:
 * instructionPercentCoverageChange :**-26.92%**
 * branchPercentCoverageChange :**-25.00%**
 * complexityPercentCoverageChange :**-25.00%**
    
org.pentaho.mongo.wrapper.NoAuthMongoClientWrapper:
 * instructionPercentCoverageChange :**-5.44%**
 * branchPercentCoverageChange :**-2.00%**
 * linePercentCoverageChange :**-4.57%**
 * complexityPercentCoverageChange :**-9.21%**
 * methodPercentCoverageChange :**-19.23%**
    
org.pentaho.mongo.wrapper.MongoClientWrapperFactory:
 * instructionPercentCoverageChange :**-11.32%**
 * linePercentCoverageChange :**-12.50%**
    
org.pentaho.mongo.wrapper.UsernamePasswordMongoClientWrapper:
 * instructionPercentCoverageChange :**-57.19%**
 * linePercentCoverageChange :**-50.38%**
 * complexityPercentCoverageChange :**-35.00%**
 * methodPercentCoverageChange :**-25.00%**
    
org.pentaho.mongo.MongoProp:
 * instructionPercentCoverageChange : +0.18%

**Comment by flatrixnet** [comment]
Fixed NPE's and Code Styles
NPE's was caused by dependency on "pentaho-mongo-utils" additional tests to avoid this problem.

**Comment by lucboudreau** [comment]
http://10.100.11.222:8080/job/wingman-osgi/71/

**Comment by lucboudreau** [comment]
Head build failed with return code 1

**Comment by lucboudreau** [comment]
Don't expect Wingman results for this one. This particular project takes many environmental tweaks. Have you guys tested this correctly on local?

**Comment by flatrixnet** [comment]
I have tested this on a mongo DB with SHA-1 Authentication and a mongo 3.2.0 with no authentication.
This pull request depends also on this:
https://github.com/pentaho/pentaho-karaf-assembly/pull/187

in the end fixes this 
https://github.com/pentaho/pentaho-mongodb-plugin/pull/120

**Comment by pamval** [comment]
Done in #51

File: build.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
       <arg value="pentaho-mongo-utils.bnd" />
       <arg value="--output" />
       <arg value="${dist.dir}/${ivy.artifact.id}-${project.revision}.jar" />
-      <arg value="${dist.dir}/${ivy.artifact.id}-${project.revision}-noosgi.jar" />
+      <arg value="${dist.dir}/${ivy.artifcd git remote -vact.id}-${project.revision}-noosgi.jar" />
```
Comment (lucboudreau) [inline build.xml]: This does not look right. Please revert this change.

Comment (lucboudreau) [inline build.xml]: Nevermind. It's gone in #51.

**Comment by eliofreitas** [comment]
@lucboudreau it was reverted in here https://github.com/pentaho/pentaho-mongo-utils/pull/51/files
